docs/cn/Example_LMQ.md

```diff
@@ -0,0 +1,75 @@
+# Light message queue (LMQ)
+
+
+## 一、broker启动配置
+
+
+broker.conf文件需要增加以下的配置项，开启LMQ开关，这样就可以识别LMQ相关属性的消息，进行原子分发消息到LMQ队列
+```properties
+enableLmq = true
+enableMultiDispatch = true
+```
+## 二、发送消息
+发送消息的时候通过设置 INNER_MULTI_DISPATCH 属性，LMQ queue使用逗号分割，queue前缀必须是 %LMQ%，这样broker就可以识别LMQ queue.
+```java
+DefaultMQProducer producer = new DefaultMQProducer("please_rename_unique_group_name");
+producer.setNamesrvAddr("name-server1-ip:9876;name-server2-ip:9876");
+producer.start();
+
+
+/*
+* Create a message instance, specifying topic, tag and message body.
+*/
+Message msg = new Message("TopicTest" /* Topic */,
+                          "TagA" /* Tag */,
+                          ("Hello RocketMQ " + i).getBytes(RemotingHelper.DEFAULT_CHARSET) /* Message body */
+                         );
+/*
+* INNER_MULTI_DISPATCH property and PREFIX must start as "%LMQ%",
+* If it is multiple LMQ, need to use “,” split
+*/
+message.putUserProperty("INNER_MULTI_DISPATCH", "%LMQ%123,%LMQ%456");
+/*
+* Call send message to deliver message to one of brokers.
+*/
+SendResult sendResult = producer.send(msg);
+```
+## 三、拉取消息
+LMQ queue在每个broker上只有一个queue，也即queueId为0， 指明轻量级的MessageQueue，就可以拉取消息进行消费。
+```java
+DefaultMQPullConsumer defaultMQPullConsumer = new DefaultMQPullConsumer();
+defaultMQPullConsumer.setNamesrvAddr("name-server1-ip:9876;name-server2-ip:9876");
+defaultMQPullConsumer.setVipChannelEnabled(false);
+defaultMQPullConsumer.setConsumerGroup("CID_RMQ_SYS_LMQ_TEST");
+defaultMQPullConsumer.setInstanceName("CID_RMQ_SYS_LMQ_TEST");
+defaultMQPullConsumer.setRegisterTopics(new HashSet<>(Arrays.asList("TopicTest")));
+defaultMQPullConsumer.setBrokerSuspendMaxTimeMillis(2000);
+defaultMQPullConsumer.setConsumerTimeoutMillisWhenSuspend(3000);
+defaultMQPullConsumer.start();
+
+String brokerName = "set broker Name";
+MessageQueue mq = new MessageQueue("%LMQ%123", brokerName, 0);
+
+Long offset = defaultMQPullConsumer.maxOffset(mq);
+
+defaultMQPullConsumer.pullBlockIfNotFound(
```

Comment:
Hi, I tested the doc and found some problems with the example:

1. producer: queueId is illegal
```
aused by: org.apache.rocketmq.client.exception.MQBrokerException: CODE: 1  DESC: request queueId[3] is illegal, TopicConfig [topicName=TopicTest, readQueueNums=1, writeQueueNums=1, perm=RW-, topicFilterType=SINGLE_TAG, topicSysFlag=0, order=false] Producer: 172.18.40.26:50188 BROKER: 172.18.40.26:10911
For more information, please visit the url, http://rocketmq.apache.org/docs/faq/
	at org.apache.rocketmq.client.impl.MQClientAPIImpl.processSendResponse(MQClientAPIImpl.java:662)
	at org.apache.rocketmq.client.impl.MQClientAPIImpl.sendMessageSync(MQClientAPIImpl.java:508)
	at org.apache.rocketmq.client.impl.MQClientAPIImpl.sendMessage(MQClientAPIImpl.java:490)
	at org.apache.rocketmq.client.impl.MQClientAPIImpl.sendMessage(MQClientAPIImpl.java:434)
	at org.apache.rocketmq.client.impl.producer.DefaultMQProducerImpl.sendKernelImpl(DefaultMQProducerImpl.java:838)
	at org.apache.rocketmq.client.impl.producer.DefaultMQProducerImpl.sendDefaultImpl(DefaultMQProducerImpl.java:581)
	... 4 more
```


2. consumer: The broker[broker-a] not exist
```
Exception in thread "main" org.apache.rocketmq.client.exception.MQClientException: The broker[broker-a] not exist
For more information, please visit the url, http://rocketmq.apache.org/docs/faq/
	at org.apache.rocketmq.client.impl.MQAdminImpl.maxOffset(MQAdminImpl.java:219)
	at org.apache.rocketmq.client.impl.consumer.DefaultMQPullConsumerImpl.maxOffset(DefaultMQPullConsumerImpl.java:170)
	at org.apache.rocketmq.client.consumer.DefaultMQPullConsumer.maxOffset(DefaultMQPullConsumer.java:155)
	at org.apache.rocketmq.example.simple.PullConsumer.main(PullConsumer.java:49)
```

Add `Thead.sleep(30000)` will solve it, because the lmq topic is not in nameserver,  should wait until the timer fetch broker names.
 
and I enhanced then consumer, to make it consume continously:

```java
DefaultMQPullConsumer defaultMQPullConsumer = new DefaultMQPullConsumer();
        defaultMQPullConsumer.setNamesrvAddr("localhost:9876");
        defaultMQPullConsumer.setVipChannelEnabled(false);
        defaultMQPullConsumer.setConsumerGroup("CID_RMQ_SYS_LMQ_TEST");
        defaultMQPullConsumer.setInstanceName("CID_RMQ_SYS_LMQ_TEST");
        defaultMQPullConsumer.setRegisterTopics(new HashSet<>(Collections.singletonList("TopicTest")));

        defaultMQPullConsumer.start();
        Thread.sleep(30000);

        String brokerName = "broker-a";
        MessageQueue mq = new MessageQueue("%LMQ%123", brokerName, 0);

        long offset = defaultMQPullConsumer.maxOffset(mq);
        while (true) {
            PullResult result = defaultMQPullConsumer.pullBlockIfNotFound(mq, "*", offset, 32);
            if (result.getMsgFoundList() == null || result.getMsgFoundList().isEmpty()) {
                continue;
            }
            for (MessageExt messageExt : result.getMsgFoundList()) {
                System.out.printf("%s%n", messageExt);
            }
            offset = result.getNextBeginOffset();
        }


```

Reply:
1. request queueId[3] is illegal, TopicConfig [topicName=TopicTest, readQueueNums=1, writeQueueNums=1, perm=RW-, topicFilterType=SINGLE_TAG, topicSysFlag=0, order=false] Producer: 172.18.40.26:50188 BROKER: 172.18.40.26:10911
For more information, please visit the url, http://rocketmq.apache.org/docs/faq/

TopicTest 多创建几个队列就可以了 这个只有一个队列 所以队列queueId 3拿不到
2. broker-a 不存在问题 就是你部署的mq集群中，没有这个broke

Reply:
1. TopicTest有3个队列，因为light队列中将队列数改为1导致判断有问题。
2. broker-a 是存在的，defaultMQPullConsumer.maxOffset中代码
```
String brokerAddr = this.mQClientFactory.findBrokerAddressInPublish(mq.getBrokerName());
        if (null == brokerAddr) {
            this.mQClientFactory.updateTopicRouteInfoFromNameServer(mq.getTopic());
            brokerAddr = this.mQClientFactory.findBrokerAddressInPublish(mq.getBrokerName());
        }
```
updateTopicRouteInfoFromNameServer是通过topic去nameserver更新broker列表，nameserver中不存在这个topic（%LMQ%123），所以无法获取broker。

Reply:
> 1. TopicTest有3个队列，因为light队列中将队列数改为1导致判断有问题。
> 2. broker-a 是存在的，defaultMQPullConsumer.maxOffset中代码
> 
> ```
> String brokerAddr = this.mQClientFactory.findBrokerAddressInPublish(mq.getBrokerName());
>         if (null == brokerAddr) {
>             this.mQClientFactory.updateTopicRouteInfoFromNameServer(mq.getTopic());
>             brokerAddr = this.mQClientFactory.findBrokerAddressInPublish(mq.getBrokerName());
>         }
> ```
> 
> updateTopicRouteInfoFromNameServer是通过topic去nameserver更新broker列表，nameserver中不存在这个topic（%LMQ%123），所以无法获取broker。

%LMQ%这种是不会上报nameserver的，根据那个需要分发的普通topic来获取，TopicTest是普通的，不能当作lmq, lmq只能内部分发，lmq topic只支持命名格式%LMQ%开头的name

Reply:
> 1. TopicTest有3个队列，因为light队列中将队列数改为1导致判断有问题。
> 2. broker-a 是存在的，defaultMQPullConsumer.maxOffset中代码
> 
> ```
> String brokerAddr = this.mQClientFactory.findBrokerAddressInPublish(mq.getBrokerName());
>         if (null == brokerAddr) {
>             this.mQClientFactory.updateTopicRouteInfoFromNameServer(mq.getTopic());
>             brokerAddr = this.mQClientFactory.findBrokerAddressInPublish(mq.getBrokerName());
>         }
> ```
> 
> updateTopicRouteInfoFromNameServer是通过topic去nameserver更新broker列表，nameserver中不存在这个topic（%LMQ%123），所以无法获取broker。**sleep 30秒等待定时器同步broker列表之后就能在缓存中拿到。**
> 
> 放出来的示例我试验的结果是这样的，不知道你有没有试过

这个地方brokerName 我不是从nameserver动态获取的，我知道lmq会分发在我的demo broker-a上，直接写死的，你的lmq能分发到哪个broker 就从哪个broker上拉取消息 这个lmq不会上报nameServer,只能根据普通topic来获取brokername列表